model/config.go

```diff
@@ -161,6 +161,32 @@ func (i *IstioRegistry) RouteRules(namespace string) []*proxyconfig.RouteRule {
 	return out
 }
 
+// RouteRulesBySource selects routing rules from source to destination (sorted by precedence)
+func (i *IstioRegistry) RouteRulesBySource(namespace string, instances []*ServiceInstance) []*proxyconfig.RouteRule {
+	out := make([]*proxyconfig.RouteRule, 0)
+	for _, rule := range i.RouteRules(namespace) {
+		if rule.Match != nil {
```

Comment:
actually, we'd have to duplicate code for rule.Match == nil. maybe leave it as is?